provider-blackduck/src/main/java/com/blackduck/integration/alert/provider/blackduck/processor/message/service/BlackDuckComponentVulnerabilityDetailsCreator.java

```diff
@@ -109,14 +107,31 @@ private boolean requiresRemediation(BlackDuckProjectVersionComponentVulnerabilit
     private AlertVulnerability toAlertVulnerabilityView(BlackDuckProjectVersionComponentVulnerabilitiesView vulnerability) {
         String name = vulnerability.getId();
         String url = vulnerability.getFirstLinkSafely("vulnerability").map(HttpUrl::toString).orElse(null);
-        String severity;
-
-        ProjectVersionComponentVersionVulnerabilityRemediationCvss3View cvss3 = vulnerability.getCvss3();
-        if (vulnerability.getUseCvss3() && null != cvss3) {
-            severity = Optional.ofNullable(cvss3.getSeverity()).map(Enum::name).orElse(VulnerabilitySeverityType.HIGH.name());
+        String severity = VulnerabilitySeverityType.HIGH.name();
+
+        if (vulnerability.getCvssVersion() != null) {
+            switch (vulnerability.getCvssVersion()) {
+                case "CVSS4":
+                    ProjectVersionComponentVersionVulnerabilityRemediationCvssView cvss4 = vulnerability.getCvss4();
+                    severity = Optional.ofNullable(cvss4.getSeverity()).map(Enum::name).orElse(VulnerabilitySeverityType.HIGH.name());
+                    break;
+                case "CVSS3":
+                    ProjectVersionComponentVersionVulnerabilityRemediationCvssView cvss3 = vulnerability.getCvss3();
+                    severity = Optional.ofNullable(cvss3.getSeverity()).map(Enum::name).orElse(VulnerabilitySeverityType.HIGH.name());
+                    break;
+                case "CVSS2":
+                    ProjectVersionComponentVersionVulnerabilityRemediationCvssView cvss2 = vulnerability.getCvss2();
+                    severity = Optional.ofNullable(cvss2.getSeverity()).map(Enum::name).orElse(VulnerabilitySeverityType.HIGH.name());
+                    break;
+            }
         } else {
-            ProjectVersionComponentVersionVulnerabilityRemediationCvss2View cvss2 = vulnerability.getCvss2();
-            severity = Optional.ofNullable(cvss2.getSeverity()).map(Enum::name).orElse(VulnerabilitySeverityType.HIGH.name());
+            if (Boolean.TRUE.equals(vulnerability.getUseCvss3()) && vulnerability.getCvss3() != null) {
```

Comment:
Should there be a case for CVSS 4 here?

Reply:
This else block is the old code which doesnt have cvss4. It's left in for backwards compatibility that Lisa requested.